src/EightBall.elm

```diff
@@ -565,11 +601,17 @@ scratch when =
 -}
 type WhatHappened
     = IllegalBreak (Pool AwaitingRack)
-    | PlayersFault (Pool AwaitingPlaceBallInHand)
+    | PlayersFault NextPlayerAction
     | NextShot (Pool AwaitingPlayerShot)
     | GameOver (Pool AwaitingStart) { winner : Player }
 
 
+type NextPlayerAction
+    = SpotEightBall (Pool AwaitingSpotEightBall)
+      -- | ChooseNextAction NextPlayerAction NextPlayerAction (List NextPlayerAction)
```

Comment:
A new variant here, `ChooseNextAction`, is the idea I have for allowing multiple actions.